### PR TITLE
Issue #3291402 by alex.ksis: Add redirect to the correct membership requests page.

### PR DESCRIPTION
**Pull request by ksis**
There are two pages with membership requests. One is provided by the `grequest` module, another one is provided by the `social_group` module.

## Problem
There are two pages with membership requests. One of them is not correct and has broken "Approve" and "Reject" links.

## Solution
Create a redirect to the correct page.

## Issue tracker
https://getopensocial.atlassian.net/browse/YANG-8012
https://www.drupal.org/project/social/issues/3291402

## How to test
- [ ] Go to to the /group/%/members-pending
- [ ] Make sure you were redirected to the /group/%/membership-requests

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Release notes
Added redirect from the "Membership requests" page provided by the grequest module to the page provided by the social group module.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2998-qtynbj2vtf1hcgxeijhxu29s0de9mz5o.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62b0509414d98b2defc15756

**Review by ribel — APPROVED** [review]
@ksis It looks fine for me, but I would like to get approval from core team if this approach looks ok to them.

I see original page comes from here: https://github.com/goalgorilla/open_social/blob/902163a97846f10967754421a67396ed159643e8/modules/custom/grequest/config/optional/views.view.group_pending_members.yml

And here we override it: https://github.com/goalgorilla/open_social/blob/6defea9cc3e5bbca8c76d84982f69192abbd5bdb/modules/social_features/social_group/modules/social_group_request/src/SocialGroupRequestConfigOverride.php

cc @Ressinel, @navneet0693

File: modules/social_features/social_group/src/EventSubscriber/RedirectSubscriber.php
Comment (Ressinel) [inline modules/social_features/social_group/src/EventSubscriber/RedirectSubscriber.php:49]: @ksis Let's add an explanation of why we add this.

**Review by Ressinel — CHANGES_REQUESTED** [review]

**Comment by Ressinel** [comment]
Hi, @ksis 
Could you check my comment above, please?

File: modules/social_features/social_group/src/EventSubscriber/RedirectSubscriber.php
Comment (ksis) [reply modules/social_features/social_group/src/EventSubscriber/RedirectSubscriber.php:49]: Hi @Ressinel. I added explanation. Please, check.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2998-orfotwrzk3cfpjolud6mw4zspio0tbmp.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62e000707e6c3f16da235461

**Comment by Ressinel** [comment]
@ksis Fix PHPCS error, please.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2998-orfotwrzk3cfpjolud6mw4zspio0tbmp.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/62e000707e6c3f16da235461

**Comment by Ressinel** [comment]
There is a conflict with `11.3.x` and below. So, 🍒-picked to `11.4.x` only.

**Pull request merged**
